### PR TITLE
[ADD] purchase_order_variant_mgmt

### PR DESCRIPTION
**Pull request by pedrobaeza**
# 
# Handle easily multiple variants on Purchase Orders

This module allows to add/modify all the variants of a product in a direct
screen without the need of handling them one by one.
# Configuration
- Configure your user to have any permission from "Purchases" group.
- Create a product with 2 attributes and several values.
# Usage
- Go to Purchases > Purchase > Requests for Quotation
- Create a new quotation or edit an existing one.
- Press "Add variants" button located in the upper right corner of the
  "Order Lines" tab.
- A new screen will appear allowing you to select the products that have
  variants.
- Once you select the product, a 2D matrix will appear with the first
  attribute values as columns and the second one as rows.
- If there are already order lines for the product variants, the current
  quantity will be pre-filled in the matrix.
- Change the quantities for the variant you want and click on "Transfer to
  order"
- Order lines for the variants will be created/removed to comply with the
  input you have done.

As extra feature for saving steps, there's also a button on each existing line
that corresponds to a variant that opens the dialog directly with the product
selected.
# Known issues / Roadmap
- The inline button for modifying quantities for an existing line won't
  work correctly until these 2 PRs are merged in Odoo:
  - https://github.com/odoo/odoo/pull/13558
  - https://github.com/odoo/odoo/pull/13635
  
  The problems are already fixed in OCB.

### CODE REVIEW

**Comment by pedrobaeza** [comment]
cc @Tecnativa

**Comment by pedrobaeza** [comment]
This is the twin module as in sales - https://github.com/OCA/sale-workflow/tree/9.0/sale_order_variant_mgmt -, but for purchases

**Review by rafaelbn — APPROVED** [review]
Tested in runbot 👍 +1

File: purchase_order_variant_mgmt/models/purchase_order.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Pedro M. Baeza <pedro.baeza@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import models, fields
```
Comment (susport) [inline purchase_order_variant_mgmt/models/purchase_order.py]: Please Order by name the import about OCA guide.

File: purchase_order_variant_mgmt/wizard/purchase_manage_variant.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Pedro M. Baeza <pedro.baeza@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import openerp.addons.decimal_precision as dp
+from openerp import api, models, fields
```
Comment (susport) [inline purchase_order_variant_mgmt/wizard/purchase_manage_variant.py]: Please order by name import about oca guide

**Comment by susport** [comment]
Runbot is 👍

**Pull request merged**